src/olympia/scanners/tasks.py

```diff
@@ -0,0 +1,80 @@
+import os
+
+import requests
+
+from django.conf import settings
+from django_statsd.clients import statsd
+
+import olympia.core.logger
+
+from olympia.amo.celery import task
+from olympia.constants.scanners import SCANNERS, CUSTOMS
+from olympia.files.models import FileUpload
+
+from .models import ScannersResult
+
+log = olympia.core.logger.getLogger('z.scanners.task')
+
+
+def run_scanner(upload_pk, scanner, api_url, api_key):
+    """
+    Run a scanner on a FileUpload via RPC and store the results.
+    """
+    scanner_name = dict(SCANNERS).get(scanner)
+    log.info('Starting scanner "%s" task for FileUpload %s.', scanner_name,
+             upload_pk)
+
+    upload = FileUpload.objects.get(pk=upload_pk)
+
+    if not upload.path.endswith('.xpi'):
+        log.info('Not running scanner "%s" for FileUpload %s, it is not a xpi '
+                 'file.', scanner_name, upload_pk)
+        return
+
+    try:
+        if not os.path.exists(upload.path):
+            raise ValueError('Path "{}" is not a file or directory or does '
+                             'not exist.' .format(upload.path))
+
+        result = ScannersResult()
+        result.upload = upload
+        result.scanner = scanner
+
+        with statsd.timer('devhub.{}'.format(scanner_name)):
+            headers = {'Authorization': 'Bearer {}'.format(api_key)}
+            with open(upload.path, 'rb') as xpi:
+                response = requests.post(url=api_url,
```

Comment:
I think we'll want to pass a `timeout` here. Technically it will matter less once this is run as a proper task, because we can just rely on the task execution time limits, but still a good thing to have, with a setting to control the value.

Reply:
Good idea, thanks